src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -769,13 +770,24 @@ void ClassFileParser::parse_constant_pool(const ClassFileStream* const stream,
             const int name_ref_index =
               cp->name_ref_index_at(name_and_type_ref_index);
             const Symbol* const name = cp->symbol_at(name_ref_index);
-            if (name != vmSymbols::object_initializer_name()) {
+
+            if (supports_inline_types() && name == vmSymbols::inline_factory_name()) { // <new>
```

Comment:
Should we add a case where the method name is a factory method name but the class file doesn't support inline types?